pkg/kubelet/container/runtime_cache.go

```diff
@@ -89,24 +97,43 @@ func (r *runtimeCache) ForceUpdateIfOlder(minExpectedCacheTime time.Time) error
 	return nil
 }
 
+// updateCacheWithLock is used for testing.
+func (r *runtimeCache) updateCacheWithLock() error {
+	r.Lock()
+	defer r.Unlock()
+	return r.updateCache()
+}
+
 func (r *runtimeCache) updateCache() error {
-	pods, err := r.getter.GetPods(false)
+	pods, timestamp, err := r.getPodsWithTimestamp()
 	if err != nil {
 		return err
 	}
-	r.pods = pods
-	r.cacheTime = time.Now()
+	r.writePodsIfNewer(pods, timestamp)
 	return nil
 }
 
+func (r *runtimeCache) getPodsWithTimestamp() ([]*Pod, time.Time, error) {
+	// Always record the timestamp before getting the pods to avoid stale pods.
+	timestamp := time.Now()
+	pods, err := r.getter.GetPods(false)
+	return pods, timestamp, err
+}
+
+func (r *runtimeCache) writePodsIfNewer(pods []*Pod, timestamp time.Time) {
+	// Writes the pods and timestamp if they are newer than the cached pods.
```

Comment:
why not just include this godoc style, and also mention that this method expects the caller to take the lock?